temporal-sdk/src/main/java/io/temporal/common/converter/DataConverter.java

```diff
@@ -108,6 +110,28 @@ <T> T fromPayloads(
       int index, Optional<Payloads> content, Class<T> valueType, Type valueGenericType)
       throws DataConverterException;
 
+  /**
+   * Instantiate an appropriate Java Exception from a serialized Failure object. The default
+   * implementation delegates the conversion process to an instance of {@link
+   * io.temporal.failure.FailureConverter}, using this data converter for payload decoding.
+   *
+   * @param failure Failure protobuf object to deserialize into an exception
+   * @throws NullPointerException if failure is null
+   */
+  @Nonnull
+  RuntimeException failureToException(@Nonnull Failure failure);
+
+  /**
+   * Serialize an existing Java Exception into a Failure object. The default implementation
+   * delegates the conversion process to an instance of {@link
+   * io.temporal.failure.FailureConverter}, using this data converter for payload encoding.
+   *
+   * @param e A Java Exception object to serialize into a Failure protobuf object
```

Comment:
I believe we don't capitalize the first letters of param sentences. Also, it's any throwable, not just an exception.